src/icu.rs

```diff
@@ -628,17 +628,20 @@ impl Regex {
     /// # Safety
     ///
     /// The caller must ensure that the given `Text` outlives the `Regex` instance.
-    pub unsafe fn set_text(&mut self, text: &Text) {
+    pub unsafe fn set_text(&mut self, text: &Text, offset: usize) {
         let f = assume_loaded();
         let mut status = icu_ffi::U_ZERO_ERROR;
         unsafe { (f.uregex_setUText)(self.0, text.0 as *const _ as *mut _, &mut status) };
+        // `uregex_setUText` resets the regex to the start of the text.
+        // Because of this, we must also call `uregex_reset64`.
+        unsafe { (f.uregex_reset64)(self.0, offset as i64, &mut status) };
```

Comment:
Does setting the offset to the start of the match cause us to deadlock if you Find "a" replace with "a"? Do we need somehow to skip the matched section?

Reply:
In this case the caller specifies the "current offset" which is past the replaced range.